src/app/core/component/cardrenderer/card-renderer.component.ts

```diff
@@ -4,33 +4,105 @@ import {ActivatedRoute, Router} from '@angular/router';
 import UriEntity from '../../model/hal/uri-entity.model';
 import Pattern from '../../model/hal/pattern.model';
 import {HalLink} from '../../model/hal/hal-link.interface';
+import {PatternService} from '../../service/pattern.service';
 
 @Component({
-    selector: 'pp-card-renderer',
-    templateUrl: './card-renderer.component.html',
-    styleUrls: ['./card-renderer.component.scss']
+  selector: 'pp-card-renderer',
+  templateUrl: './card-renderer.component.html',
+  styleUrls: ['./card-renderer.component.scss']
 })
 export class CardRendererComponent {
 
-    @Input() uriEntities: Array<Pattern>;
-    @Input() showLinks = true;
+  @Input() uriEntities: Array<Pattern>;
+  @Input() showLinks = true;
     @Output() createEntityClicked: EventEmitter<void> = new EventEmitter<void>();
 
-    constructor(private zone: NgZone,
-                private router: Router,
-                private activatedRoute: ActivatedRoute) {
-    }
+  constructor(private zone: NgZone,
+              private router: Router,
+              private activatedRoute: ActivatedRoute,
+              private patternService: PatternService) {
+  }
 
-    navigate(pattern: UriEntity): void {
-        this.zone.run(() => {
-            this.router.navigate([UriConverter.doubleEncodeUri(pattern.id)], {relativeTo: this.activatedRoute});
-        });
-    }
+  navigate(pattern: UriEntity): void {
+    this.zone.run(() => {
+      this.router.navigate([UriConverter.doubleEncodeUri(pattern.id)], {relativeTo: this.activatedRoute});
+    });
+  }
 
-    getLinkCount(directedEdges: HalLink[] | HalLink): number {
-        if (!directedEdges) {
-            return 0;
+  delete(pattern: Pattern): void {
+    this.patternService.deletePattern(pattern._links.self.href)
+      .subscribe(
+        value => this.handlePatternDelete(pattern),
+        error => {
+          console.log(`Delete unsuccessful! ${error}`);
         }
-        return Array.isArray(directedEdges) ? directedEdges.length : 1;
+      );
+  }
+
+  getLinkCount(directedEdges: HalLink[] | HalLink): number {
+    if (!directedEdges) {
+      return 0;
+    }
+    return Array.isArray(directedEdges) ? directedEdges.length : 1;
+  }
+
+  /**
+   * This method is called when the pattern is successfully deleted by the pattern service.
+   * It removes the selected pattern from the uriEntities, and
+   * removes all edges from all other patterns where the pattern to delete
+   * is part of the edge.
+   * This method is only used to update the rendered information in the frontend.
+   *
+   * @param pattern: The pattern which is deleted.
+   */
+  private handlePatternDelete(pattern: Pattern): void {
+    this.uriEntities = this.uriEntities.filter(value => value !== pattern);
```

Comment:
Just a minor comment: I'd adjust the filter check to compare the ids instead of the entities